test-lit/unit/version_test.f90

```diff
@@ -1,4 +1,4 @@
 ! Check that lit configuration works by checking the compiler version
 
-! RUN: f18 -V 2>&1 | FileCheck  -check-prefix=VERSION %s
-! VERSION-DAG: f18 compiler
+! RUN: %f18 -V 2>&1 | FileCheck  -check-prefix=VERSION %s
+! VERSION: f18 compiler
```

Comment:
When I run this I get

> f18 compiler (under development)

Presumably this test is a demo of what the new framework can do rather than some placeholder for a real test. Therefore I think we may as well make this first lit test a proper test of functionality. -V is a great candidate and I expect that in the fullness of time the output might have some sort of sensible version number in it that we might check the format of with a regex in our test. For now it says this message, so regex is too much, but I think we should match the whole string in this test to be sure and also check there is no additional output after this line.

Put it another way, of we accidentally changed the message to something silly like "f18 compiler (foo bar wibble)" then this test wouldn't catch that for us would it?

Reply:
Done!